typescript/listeners/format.py

```diff
@@ -24,8 +24,10 @@ def on_modified_with_info(self, view, info):
             info.prev_sel[0].empty()
         ):
             last_command, args, repeat_times = view.command_history(0)
+            redo_command = view.command_history(1)[0]
             log.debug("last_command:{0}, args:{1}".format(last_command, args))
-            if last_command == "insert":
+            log.debug("redo_command:{0}".format(redo_command))
+            if last_command == "insert" and redo_command != "typescript_format_on_key":
```

Comment:
The original assumption was that every insert command is followed by a ``format_on_key`` if enabled. Though on a second thought, checking what the redo command is seems to be redundant.